### PR TITLE
Edl/tangent booleans

### PR DESCRIPTION
**Pull request by EarlinLutz**
Prior implementation of area booleans was incomplete.
  * when sorting outgoing edges "around a vertex", sort logic needs to consider curvature as secondary key among common angles.
  * when splitting curves for intersection points, partial interval from 0 to first intersection was missing.

### CODE REVIEW

File: core/geometry/src/test/curve/RegionBoolean.test.ts [hunk 1]
```diff
@@ -779,11 +898,48 @@ describe("GeneralSweepBooleans", () => {
...
     for (const yB of [-0.5, 0.5, 6.5, 7.5, 3.0]) {
       const rectangle2 = Loop.create(LineString3d.create(Sample.createRectangle(5, yB, 6, yB + 1, 0, true)));
-      exerciseAreaBooleans([region], [rectangle2], ck, allGeometry, x0 += dx, 0);
+      exerciseAreaBooleans([region], [rectangle2], ck, allGeometry, x0 += dx, 0, false);
     }
     GeometryCoreTestIO.saveGeometry(allGeometry, "sweepBooleans", "HoleInA");
     expect(ck.getNumErrors()).equals(0);
   });
+  it.only("FullCircle", () => {
+    const ck = new Checker();
```
Comment (dassaf4) [inline core/geometry/src/test/curve/RegionBoolean.test.ts:907]: remove it.only

File: core/geometry/src/test/curve/RegionBoolean.test.ts [hunk 2]
```diff
@@ -436,23 +440,138 @@ describe("RegionBoolean", () => {
...
-
-      // explicitly verify known output
-      const result = RegionOps.regionBooleanXY(inputs[0], inputs[1], RegionBinaryOpType.Union);
-      const output = IModelJson.Reader.parse(JSON.parse(fs.readFileSync("./src/test/testInputs/curve/areaBoolean/AreaBoolean.716145.Output.imjs", "utf8"))) as AnyRegion;
-      if (output !== undefined && result !== undefined)
-        ck.testTrue(output.isAlmostEqual(result), "Area Boolean union result as expected");
-
-      expect(ck.getNumErrors()).equals(0);
+        // Near-tangency union from MS Bug#716145. TODO: CURRENTLY FAILS.
+  it.only("NearTangencyUnion", () => {
```
[the remark this answers is not in the document]
Comment (EarlinLutz) [reply core/geometry/src/test/curve/RegionBoolean.test.ts]: (Corrected now)

File: core/geometry/src/test/curve/RegionBoolean.test.ts (hunk 1 above)
Comment (EarlinLutz) [reply core/geometry/src/test/curve/RegionBoolean.test.ts:907]: (corrected now)

File: core/geometry/src/topology/Merging.ts [hunk 3]
```diff
@@ -327,7 +430,8 @@ export class HalfEdgeGraphMerge {
               // NO leave nodeA and thetaA   ignore nodeB -- later step will get the outside of its banana.
             } else {
               HalfEdge.pinch(nodeA, nodeB);
-              if (Angle.isAlmostEqualRadiansAllowPeriodShift(thetaA, thetaB)) {
+              const doNullFaceMarkup = false;
+              if (doNullFaceMarkup && Angle.isAlmostEqualRadiansAllowPeriodShift(thetaA, thetaB)) {
```
Comment (dassaf4) [inline core/geometry/src/topology/Merging.ts:436]: Does the new secondary sort on curvature obviate the need to markup null faces?

**Review by dassaf4 — APPROVED** [review]

**Pull request merged**